temfpy/optimization.py

```diff
@@ -179,3 +179,77 @@ def rosenbrock(x):
     rslt = rosen(x)
 
     return rslt
+
+
+def carlberg(x, a, b):
+    r"""Carlberg function.
+
+    .. math::
+        f(x) = \frac{1}{2}\sum_{i=1}^n a_i (x_i - 1)^2 + b \left[n -
+        \sum_{i=1}^n \cos(2 \pi(x_i-1)) \right]
+
+    Parameters
+    ----------
+    x : array_like
+        Input vector with dimension :math:`d`.
+    a : array_like
+        Input vector with dimension :math:`d`.
+    b : float
+        Must not be smaller than zero.
+        For more information see Notes.
+
+    Returns
+    -------
+    float
+         Output domain
+
+    Notes
+    -----
+    If the values in :math:`a` are widely distributed the function is
+    said to be ill-conditioned and it is hard to minimize in some
+    directions for Hessian free numerical methods.
+    If :math:`b=0` (see second graph below) the function is
+    convex, smooth and has its minimum at
+    :math:`x = \begin{pmatrix}1 & \dots & 1 \end{pmatrix}^T`. For
+    :math:`b>0` the function is no longer convex and has many local
+    minima (see first graph below), making it hard for
+    local optimization methods to find the global minimum,
+    which is still at
+    :math:`x = \begin{pmatrix}1 & \dots & 1 \end{pmatrix}^T`.
+
+    .. figure:: ../../docs/_static/images/fig-carlberg_noise.png
+       :align: center
+    .. figure:: ../../docs/_static/images/fig-carlberg_no_noise.png
+       :align: center

```

Comment:
Maybe you can add a title for both plots? You can infer which one is which from the notes, but titles might still be even clearer.